bam_controls.js

```diff
@@ -233,21 +233,21 @@ class BamControls extends HTMLElement {
     async connectedCallback() {
         this.broker = getDataBroker(this);
         if (this.broker) {
-            const headerPromise = new Promise((resolve, reject) => {
-                this.broker.addEventListener('header', (evt) => {
-                  resolve(evt.detail);
-                });
-              });
+            this.broker.addEventListener('alignment-data', (event) => {
+                const { header } = event.detail;
+                this.bamHeader = header;
+            });
 
             this.broker.addEventListener('stats-stream-data', (evt) => {
                 const stats = evt.detail;
                 this.total_reads = stats["total_reads"] / 1000;
 
                 this.updateSampleReads();
             });
-        
-            this.bamHeader = await headerPromise;
-            this.build = this.bamHeader[0].length === 249250621 ? 'GRCh37' : 'GRCh38';
+
+            if (this.bamHeader) {
```

Comment:
This looks like a race condition. What's going to happen if the header isn't ready?

Reply:
I think you're right! We cannot guarantee that the event is fired and the bamHeader is ready. The build is possibly `undefined`.

I will move the code into the listener together.